src/components/application_manager/src/rpc_protection_mediator_impl.cc

```diff
@@ -60,9 +61,29 @@ bool RPCProtectionMediatorImpl::DoesRPCNeedEncryption(
       encrypted_need |= rpc_encryption_manager.GroupNeedEncryption(group);
     }
   }
+
+  encrypted_need &= rpc_encryption_manager.AppNeedEncryption(policy_app_id);
```

Comment:
as far as i understand if we get `encryption_needed = false` for the application level, we don't need to check other policy levels for this `function_id`. if my understanding is correct, does it make sense to move this check upwards?